core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3141,8 +3152,12 @@ private void mountInternal(RpcContext rpcContext, LockedInodePath inodePath, All
       loadMetadataSucceeded = true;
     } finally {
       if (!loadMetadataSucceeded) {
-        mMountTable.delete(rpcContext, inodePath.getUri(), true);
+        mMountTable.delete(rpcContext, inodePath, true);
+      } else {
+        // here, the inodePath is sure to be fullPathExists.
+        mMountTable.addMountPointIntoMountTableTrie(inodePath);
       }
+      mMountTable.enableMountTableTrie();
```

Comment:
After calling `enableMountTableTrie`, if the `MountTableTrie` is `PENDING`, it will be `ENABLED`